torch/fx/experimental/proxy_tensor.py

```diff
@@ -487,22 +485,20 @@ def dispatch_trace(
 
 @contextlib.contextmanager
 def _pop_proxy_mode_temporarily(dk):
-    # This is a shim around the existng per-dispatch-key-mode logic.
-    # I'll delete the per-dispatch-key-mode logic in a followup PR
-    if dk is not None:
-        # During pre_dispatch, pop off of the PreDispatch mode stack
-        old = _pop_mode(dk)
-        try:
-            yield old
-        finally:
-            _push_mode(old, dk)
-    else:
-        # During normal tracing, pop off of the dedicated proxy mode stack
-        old = torch._C._unset_dispatch_mode(torch._C._TorchDispatchModeKey.PROXY)
-        try:
-            yield old
-        finally:
-            torch._C._set_dispatch_mode(old)
+    assert dk is None or dk == torch._C.DispatchKey.PreDispatch
+    # During normal tracing, pop off of the dedicated proxy mode stack
+    from torch._ops import unset_mode_pre_dispatch
+    from torch._ops import _set_mode_pre_dispatch
+
+    old = (
+        unset_mode_pre_dispatch(torch._C._TorchDispatchModeKey.PROXY) if dk is not None
+        else torch._C._unset_dispatch_mode(torch._C._TorchDispatchModeKey.PROXY)
```

Comment:
There are 3-4 places throughout this file where we need to switch between the set/unset for the Python stack vs the PreDispatch stack. Add a helper like `ProxyTorchDispatchMode.[un]set_proxy_mode(mode)` that does it?

Reply:
Done